File: leetcode/1.two_sum/python3/two_sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ret = []
        
        hashdict = {}
        
        for itx, x in enumerate(nums, start=0):
            hashdict[x] = itx
            
        print('hashdict: ', hashdict)
        for itx, x in enumerate(nums, start=0):
            compen = target - x
            if (compen in hashdict) and (itx != hashdict[compen]):
                print('compen: ', compen)
                ret = [itx, hashdict[compen]]
                return ret
            
#        return ret

#     def twoSum(self, nums: List[int], target: int) -> List[int]:
#         ret = []
#         for itx, x in enumerate(nums, start=0):
#             for ity, y in enumerate(nums[itx + 1:], start=itx+1):
# #                print("nums[itx + 1:]: ", nums[itx + 1:])
# #                print("itx:", itx)
#                 if target == (x + y):
#                     ret=[itx, ity]
#         return ret

File: leetcode/1.two_sum/js/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    numsMap = {}
    
    for (let p = 0; p < nums.length; p++) {
        const currentMapVal = numsMap[nums[p]];
        
        if (currentMapVal >= 0) {
            return [currentMapVal, p]
        } else {
            const numbertofind = target - nums[p]
            numsMap[numbertofind] = p
        }
    }
    
    return null;
    
};

